### PR TITLE
Convert Asyncify fake globals to fake calls. NFC

### PR DESCRIPTION
**Pull request by kripken**
Also add more clarifying comments.

### CODE REVIEW

File: src/passes/Asyncify.cpp [hunk 1]
```diff
@@ -298,47 +298,68 @@ enum class DataOffset { BStackPos = 0, BStackEnd = 4 };
...
     Builder builder(module);
-    for (auto& pair : map) {
-      auto type = pair.first;
-      auto name = pair.second;
-      rev[name] = type;
-      module.addGlobal(builder.makeGlobal(
-        name, type, LiteralUtils::makeZero(type, module), Builder::Mutable));
+    std::string prefix = "__asyncify_fake_call";
+    for (auto type : {Type::i32, Type::i64, Type::f32, Type::f64}) {
+      auto typedPrefix = prefix + '_' + Type(type).toString() + '_';
```
Comment (tlively) [inline src/passes/Asyncify.cpp]: Nit: Can we move the final `'_'` to the "get" and "set" strings below? I think that would be slightly clearer and more consistent.

File: src/passes/Asyncify.cpp [hunk 2]
```diff
@@ -1008,17 +1031,34 @@ struct AsyncifyLocals : public WalkerPass<PostWalker<AsyncifyLocals>> {
   AsyncifyLocals(ModuleAnalyzer* analyzer) : analyzer(analyzer) {}
 
   void visitCall(Call* curr) {
+    // Check if this is one of our fake calls.
+    if (analyzer->fakeCalls.getSetter(curr->target)) {
```
Comment (tlively) [inline src/passes/Asyncify.cpp]: I don't understand why this is in an `if`. If the target is not in the map, `getSetter` will throw an exception, not return an error value.

Comment (kripken) [reply src/passes/Asyncify.cpp]: `getSetter` of a function name will return the function, or nullptr if it's not a setter. (`getSetter` of a type is guaranteed to succeed, though.)

File: src/passes/Asyncify.cpp (hunk 1 above)
Comment (kripken) [reply src/passes/Asyncify.cpp]: Sure, changing.

**Comment by kripken** [comment]
Review comment addressed (tests disabled as master is broken atm).

File: src/passes/Asyncify.cpp (hunk 2 above)
Comment (tlively) [reply src/passes/Asyncify.cpp]: Oh, right you are. I was looking at the wrong overload 😅

**Pull request merged**